tests/test_docs_admonition_inserter.py

```diff
@@ -0,0 +1,198 @@
+                telegram.ResidentialAddress,  # mentioned on the second line of docstring of .data
+                ":attr:`telegram.EncryptedPassportElement.data`",
+            ),
+            (
+                "returned_in",
+                telegram.StickerSet,
+                ":meth:`telegram.Bot.get_sticker_set`",
+            ),
+            (
+                "returned_in",
+                telegram.ChatMember,
+                ":meth:`telegram.Bot.get_chat_member`",
+            ),
+            (
+                "returned_in",
+                telegram.ChatMemberOwner,
+                ":meth:`telegram.Bot.get_chat_member`",  # subclass
+            ),
+            (
+                "returned_in",
+                telegram.Message,
+                ":meth:`telegram.Bot.edit_message_live_location`",  # Union[Message, bool]
+            ),
+            (
+                "use_in",
+                telegram.InlineQueryResult,
+                ":meth:`telegram.Bot.answer_web_app_query`",  # ForwardRef
+            ),
+            (
+                "use_in",
+                telegram.InputMediaPhoto,
+                ":meth:`telegram.Bot.send_media_group`",  # Sequence[Union[...]]
+            ),
+            (
+                "use_in",
+                telegram.MaskPosition,
+                ":meth:`telegram.Bot.add_sticker_to_set`",  # optional
+            ),
+            (
+                "use_in",
+                telegram.Sticker,
+                ":meth:`telegram.Bot.get_file`",  # .file_id with lots of piped types
+            ),
+            (
+                "use_in",
+                telegram.ext.BasePersistence,
+                ":meth:`telegram.ext.ApplicationBuilder.persistence`",
+            ),
+            ("use_in", telegram.ext.Defaults, ":meth:`telegram.ext.ApplicationBuilder.defaults`"),
+            (
+                "use_in",
+                telegram.ext.JobQueue,
+                ":meth:`telegram.ext.ApplicationBuilder.job_queue`",  # TypeVar
+            ),
+            (
+                "use_in",
+                telegram.ext.PicklePersistence,  # subclass
+                ":meth:`telegram.ext.ApplicationBuilder.persistence`",
+            ),
+        ),
+    )
+    def test_check_presence(self, admonition_inserter, admonition_type, cls, link):
+        """Checks if a given link is present in the admonition of a given type for a given
+        class.
+        """
+        admonitions = admonition_inserter.admonitions
+
+        assert cls in admonitions[admonition_type]
+
+        # exactly one of the lines in the admonition for this class must consist of the link
+        # (this is a stricter check than just checking if the entire admonition contains the link)
+        lines_with_link = [
+            line
+            for line in admonitions[admonition_type][cls].splitlines()
+            # remove whitespaces and occasional bullet list marker
+            if line.strip().removeprefix("* ") == link
```

Comment:
`removeprefix` is py3.9+ only, which gives more reason to not have this run with the regular test suite

Reply:
well yes, the whole admonition generator is 3.9+. Ok, let me figure out how it's done :)